zerver/lib/export.py

```diff
@@ -2434,14 +2429,11 @@ def get_analytics_config() -> Config:
     return analytics_config
 
 
-def get_consented_user_ids(consent_message_id: int) -> set[int]:
+def get_consented_user_ids(realm: Realm) -> set[int]:
     return set(
-        Reaction.objects.filter(
-            message_id=consent_message_id,
-            reaction_type="unicode_emoji",
-            # outbox = 1f4e4
-            emoji_code="1f4e4",
-        ).values_list("user_profile", flat=True)
+        UserProfile.objects.filter(
+            realm=realm, is_active=True, is_bot=False, allow_private_data_export=True
+        ).values_list("id", flat=True)
```

Comment:
Probably the acting user should always be included, but maybe we want to do that at the UI layer, not here, so this is probably fine.